RecoLocalCalo/EcalRecProducers/plugins/EcalRecHitProducer.cc

```diff
@@ -80,13 +95,17 @@ void EcalRecHitProducer::produce(edm::Event& evt, const edm::EventSetup& es) {
 
   // get the barrel uncalib rechit collection
 
-  evt.getByToken(ebUncalibRecHitToken_, pEBUncalibRecHits);
-  ebUncalibRecHits = pEBUncalibRecHits.product();
-  LogDebug("EcalRecHitDebug") << "total # EB uncalibrated rechits: " << ebUncalibRecHits->size();
+  if (doEB_) {
+    evt.getByToken(ebUncalibRecHitToken_, pEBUncalibRecHits);
+    ebUncalibRecHits = pEBUncalibRecHits.product();
```

Comment:
what about
```suggestion
    ebUncalibRecHits = evt.get(ebUncalibRecHitToken_);
```
here and below?

Reply:
I meant to keep the changes to this producer to the minimum but if the performance gain justifies the change I can do that.

Reply:
no physics change is expected, but do you see a reason to keep using the "outdated" `getByToken`?

Reply:
By performance gain I was thinking about computing performance. Since apparently the evt.get() method is faster.

Reply:
Sure, what I meant was: if there is no physics change by construction, and if the old way is expected to be marginally slower (I don't expect it to be measurable in any case), is there any reason to keep the old way?

Reply:
I was actually not aware to the new way before. Changed to it now.